extensions/deploy_replace_undeploy/standard/requirements/cwl/package/REQ_response-body.adoc

```diff
@@ -0,0 +1,10 @@
+[[req_cwl_package_response-body]]
+[requirement]
+====
+[%metadata]
+label:: /req/cwl/package/response-body
+part:: A response with HTTP status code `200` SHALL include a body that contains:
+ * the <<rc_cwl,CWL>> to use to deploy the process, in case the Content-Type used to deploy the process was `application/cwl`. 
+ * the <<rc_ogcapppkg,OGC Application Package>> to use to deploy the process, in case the Content-Type used to deploy the process was `application/ogcapppkg+json`.
```

Comment:
Should we consider the possibility that a package deployed one way could provide the representation of another encoding? For example, if deployed using CWL, it would be fairly easy to extract the relevant parts of the document to form the corresponding *OGC Application Package* format. An `Accept` header could allow the selection of one, an if conversion is supported, could help make instances more interoperable. If so, sentences mentioning that "if deployed with X, SHOULD support X" would have to be reworded slightly.

Or, do we want to make it explicit that the original format for deployment must be returned (no conversion permitted)? In such case, the `Accept` header is irrelevant.

Reply:
I vote yes, the implementation should be able to decide which package it can return back.

Reply:
I also favor having the capability to choose between various formats. I agree that CWL (and maybe other encoding. supported in the future) can be converted into an OGC Application Package and vice versa. Nevertheless, I wonder if the `Accept` header should not be added to the `6.6.1` section, which describes the operation for retrieving the formal description.

Then, in the `6.6 Retrieving the formal description` section, we would add another section: `Exception` specifying what happens if the `Accept` header is not supported. 

The `unsupported-media-type` exception type is possible if the media type is not supported. However, it's important to note that the exception could be something else if the process cannot be exposed in the desired encoding. For instance, consider a Docker image used to publish a process as an OGC Application Package, then a request `/processes/{processId}/package` using `Accept` header set to `application/cwl` should get back an exception that would help the client application determine that the issue comes from the fact that the conversion is impossible (maybe a type set to `http://www.opengis.net/def/exceptions/ogcapi-processes-2/1.0/incompatible-media-type`).

What do you think?

Reply:
SWG meeting from April 29th: Add a note that you can request the application package in a format other than originally used using the accept header. Examples: If `cwl` is explicitly requested and the server cannot handle it that a `HTTP 406` should be returned.